qiita_db/metadata_template/sample_template.py

```diff
@@ -221,9 +250,15 @@ def extend(self, md_template):
         md_template = self._clean_validate_template(md_template, self.study_id,
                                                     SAMPLE_TEMPLATE_COLUMNS)
 
-        self._add_common_extend_steps_to_queue(md_template, conn_handler,
-                                               queue_name)
+        new_cols, new_samples = self._add_common_extend_steps_to_queue(
+            md_template, conn_handler, queue_name)
+
         conn_handler.execute_queue(queue_name)
+        self._update_analyses(self.id)
+        if new_cols:
+            self.log_change("Columns added: %s" % ', '.join(new_cols))
+        if new_samples:
```

Comment:
That's the first notice that I have for this. If the study is public, you cannot generate new ProcessedData. Note that what we are making public is the ProcessedData not the study... so there is no way of doing this...